### PR TITLE
Fixing test CI pipeline

### PR DESCRIPTION
**Pull request by Heavyblade**
### Summary:

This PR enables the test pipeline on Github by removing the Gemfiles that points to Rails master and that creates a dependency issue related with Ruby 3.

### Unit Tests:
Finished in 35.615298 seconds.
5846 tests, 79285 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

### RuboCop:
792 files inspected, no offenses detected

### CODE REVIEW

File: lib/active_merchant/billing/gateways/rapyd.rb [hunk 1]
```diff
@@ -334,7 +334,7 @@ def commit(method, action, parameters)
         )
       rescue ActiveMerchant::ResponseError => e
         response = e.response.body.present? ? parse(e.response.body) : { 'status' => { 'response_code' => e.response.msg } }
-        message = response['status'].slice('message', 'response_code').values.compact_blank.first || ''
+        message = response['status'].slice('message', 'response_code').values.select(&:present?).first || ''
```
Comment (aenand) [inline lib/active_merchant/billing/gateways/rapyd.rb:337]: Was this a rubocop autocorrect?

Comment (Heavyblade) [reply lib/active_merchant/billing/gateways/rapyd.rb:337]: No @aenand , this was a change from my side, `compact_blank` is available from Rails 6.1, so I didn't want t break rails 5.0 tests and that code is mine, so I know the intention and why `select(&:present?)` achieves the same result.

**Comment by naashton** [comment]
All seems good from an AM standpoint. It looks like this change (other than the 1 change to rapyd.rb) only impacts the CI pipeline for this repo?

**Comment by Heavyblade** [comment]
Yes @naashton this PR changes two things:

1) Address the Ruby3-rails-master issue that brakes de CI
2) removes `compact_blank` the gem can properly work on Rails 5

**Comment by almalee24** [comment]
@Heavyblade can you run this against core just to make sure everything looks good

**Comment by Heavyblade** [comment]
@almalee24 do you mean, running Rapyd and Shift4 transactions (integration specs) on CORE to ensure everything looks ok ?

**Comment by almalee24** [comment]
> @almalee24 do you mean, running Rapyd and Shift4 transactions (integration specs) on CORE to ensure everything looks ok ?

Yes but also just script/test. I think the changes are fine but just to make sure the Core will run smoothly with these updates.

**Comment by Heavyblade** [comment]
@almalee24 CORE unit tests are OK => https://github.com/spreedly/core/pull/9040

**Rapyd AM Remote Tests =>** 

```bash
Finished in 244.007879 seconds.
53 tests, 152 assertions, 2 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
96.2264% passed
```

*Note*: The 2 failures are wallet related or account related issues.

**Core Rapyd Remote Tests**:
```bash
❯ bundle exec rspec --tag remote spec/requests/v1/gateway_integrations/rapyd_spec.rb
Finished in 1 minute 21.6 seconds (files took 1 minute 17.27 seconds to load)
19 examples, 0 failures
```

**Review by almalee24 — APPROVED** [review]

**Comment by almalee24** [comment]
Thanks for working on this!!

**Pull request merged**